### PR TITLE
feat: bump to nixpkgs release 23.11

### PR DESCRIPTION
**Pull request by mkenigs**
Bumps several things to nixpkgs release 23.11:
- The ref of nixpkgs we use to build flox.
- The ref of nixpkgs we use by default to provide packages for the CLI.
- The revision we use in our tests.

An overlay was added to continue using bear from 23.05, because it is
broken on darwin in 23.11.

The bulk of these changes are changes to the tests. We use 3 different
revs across all of our test suites:
- One from release 23.11
- One from release 23.05
- A revision old enough to provide glibc 2.34

A non-exhaustive list of some of the changes that are non-obvious:
- Introduce NIXPKGS_REV_OLD in setup_suite instead of deducing OTHER_REV
  from lockfiles.
- Drop propagated packages test because it was already skipped.
- Use linux instead of zfs-kernel for prefer pre-release test because
  zfs-kernel no longer has a prerelease.
- Remove some unused test files in pkgdb.

### CODE REVIEW

File: cli/tests/package-show.bats [hunk 1]
```diff
@@ -209,26 +211,26 @@ teardown() {
   run --separate-stderr jq -r '.registry.inputs.nixpkgs.from.rev' \
     "$PROJECT_DIR/.flox/env/manifest.lock"
   assert_success
-  assert_output "$PKGDB_NIXPKGS_REV_NEW"
+  assert_output "$PKGDB_NIXPKGS_REV_OLD"
   unset output
 
   # Search for a package with `pkgdb`
   run --separate-stderr sh -c \
-    "_PKGDB_GA_REGISTRY_REF_OR_REV='$PKGDB_NIXPKGS_REV_NEW'       \
```
Comment (ysndr) [inline cli/tests/package-show.bats:217]: is this deliberately changing from "new" to "old"

File: pkgdb/tests/data/buildenv/lockfiles/propagated/manifest.toml
Comment (ysndr) [inline pkgdb/tests/data/buildenv/lockfiles/propagated/manifest.toml:1]: this is part of the propagated packages test, which .. we skip?
Need to have a look at that again, but i dont think we should delete the test data for it.

File: pkgdb/tests/setup_suite.bash [hunk 2]
```diff
@@ -83,14 +83,37 @@ misc_vars_setup() {
...
+
+  # See reasons for choosing this rev in cli tests
+  # Incomplete notes on steps needed to bump:
+  # _PKGDB_GA_REGISTRY_REF_OR_REV=$NIXPKGS_REV_OLDER \
+  #   pkgdb manifest lock --ga-registry --manifest pkgdb/tests/harnesses/proj1/manifest.toml \
+  #   | jq > pkgdb/tests/harnesses/proj1/manifest_old.lock
+  NIXPKGS_REV_OLDER="bc01a2be500c10f1507dcc8e98c9f5bd72c02aa3"
+  export NIXPKGS_REV_OLDER
+  NODEJS_VERSION_OLDEST="16.16.0"
+  export NODEJS_VERSION_OLDEST
```
Comment (ysndr) [inline pkgdb/tests/setup_suite.bash:114]: should this be "older" too, given that it comes from "nixpkgs_older"?
This older stuff is not the clearest tbh..

File: pkgdb/tests/setup_suite.bash [hunk 3]
```diff
@@ -83,14 +83,37 @@ misc_vars_setup() {
...
+  # _PKGDB_GA_REGISTRY_REF_OR_REV=$NIXPKGS_REV \
+  #   pkgdb manifest lock --ga-registry --manifest pkgdb/tests/harnesses/proj1/manifest.toml \
+  #   | jq > pkgdb/tests/harnesses/proj1/manifest.lock
+  NIXPKGS_REV="ab5fd150146dcfe41fda501134e6503932cc8dfd"
   NIXPKGS_REF="github:NixOS/nixpkgs/$NIXPKGS_REV"
 
-  NIXPKGS_FINGERPRINT="5fde12e3424840cc2752dae09751b09b03f5a33"
-  NIXPKGS_FINGERPRINT="${NIXPKGS_FINGERPRINT}c3ec4de672fc89d236720bdc7"
+  NIXPKGS_FINGERPRINT="9bb3d4c033fbad8efb5e28ffcd1d70383e0c5bb"
+  NIXPKGS_FINGERPRINT="${NIXPKGS_FINGERPRINT}cb7cc5c526b824524467b19b9"
```
Comment (ysndr) [inline pkgdb/tests/setup_suite.bash:94]: suggestion: maybe a short comment how the fingerprints are generated. 
-- still look kinda arbitrary

File: pkgdb/tests/setup_suite.bash [hunk 4]
```diff
@@ -83,14 +83,37 @@ misc_vars_setup() {
...
+  NIXPKGS_REV_OLD="e8039594435c68eb4f780f3e9bf3972a7399c4b1"
+  export NIXPKGS_REV_OLD
+  NODEJS_VERSION_OLD="18.16.0"
+  export NODEJS_VERSION_OLD
+
+  # See reasons for choosing this rev in cli tests
+  # Incomplete notes on steps needed to bump:
+  # _PKGDB_GA_REGISTRY_REF_OR_REV=$NIXPKGS_REV_OLDER \
+  #   pkgdb manifest lock --ga-registry --manifest pkgdb/tests/harnesses/proj1/manifest.toml \
+  #   | jq > pkgdb/tests/harnesses/proj1/manifest_old.lock
```
Comment (ysndr) [inline pkgdb/tests/setup_suite.bash:110]: that "older" revision was added to test behavior of ld_audit and old/incompatible glibc's.
How is that relevant in pkgdb tests? If it's not it probably shouldn't refer to the docs in `cli/tests`.

File: pkgdb/tests/test.hh [hunk 5]
```diff
@@ -23,13 +23,16 @@
 /* -------------------------------------------------------------------------- */
```
Comment (ysndr) [inline pkgdb/tests/test.hh:24]: // same as NIXPKGS_REV in ./setup_suite.bash

File: pkgdb/tests/test.hh [hunk 6]
```diff
@@ -23,13 +23,16 @@
...
 
 static const std::string nixpkgsRev
-  = "e8039594435c68eb4f780f3e9bf3972a7399c4b1";
+  = "ab5fd150146dcfe41fda501134e6503932cc8dfd";
 
 static const std::string nixpkgsRef
-  = "github:NixOS/nixpkgs/e8039594435c68eb4f780f3e9bf3972a7399c4b1";
+  = "github:NixOS/nixpkgs/ab5fd150146dcfe41fda501134e6503932cc8dfd";
```
Comment (ysndr) [inline pkgdb/tests/test.hh:29]: suggestion: reuse variable
IIRC, you should be able to concat const strings in c++

```suggestion
  = "github:NixOS/nixpkgs/" + nixpkgsRev;
```

**Review by ysndr — APPROVED** [review]
Seems pretty complete, left some suggestion to add a few more comments or reuse variables but no major issues

**Review by tomberek — APPROVED** [review]
With additional local testing and CI this looks reasonable.

File: pkgdb/tests/data/buildenv/lockfiles/propagated/manifest.toml
Comment (mkenigs) [reply pkgdb/tests/data/buildenv/lockfiles/propagated/manifest.toml:1]: Yeah it was skipped on all systems so I just dropped it. Maybe we need to fix the test and unskip it?

File: pkgdb/tests/setup_suite.bash (hunk 2 above)
Comment (mkenigs) [reply pkgdb/tests/setup_suite.bash:114]: Oh whoops I meant for that to be OLDER
